mmv1/templates/terraform/custom_delete/storage_folder_delete.go.tmpl

```diff
@@ -0,0 +1,88 @@
+bucket := d.Get("bucket").(string)
+name := d.Get("name").(string)
+
+var listError, deleteObjectError error
+for deleteObjectError == nil {
+    res, err := config.NewStorageClient(userAgent).Objects.List(bucket).Prefix(name).Do()
+    if err != nil {
+        log.Printf("Error listing contents of bucket %s: %v", bucket, err)
+        listError = err
+        break
+    }
+
+    if len(res.Items) == 0 {
+        break // 0 items, folder empty
+    }
+
+    if !d.Get("force_destroy").(bool) {
```

Comment:
I would recommend moving this to the beginning, since you should be able to escape early

Reply:
moved the logic to starting

Reply:
we first check if the folder contains any objects, if no objects we delete the folder even force_destory set to false, if any objects are there force_destory should be set to true then we delete the objects and folders
 
if force_destroy is true non-empty folders will be deleted, if force_destory is false empty folders are deleted so first we need to check if any objects present in the folders

Reply:
Ah, thanks for clarifying, I was mixing this up with how `prevent_destroy` would be working.